Project2Store/FWC/app/src/main/java/com/colinbradley/fwc/MainActivity.java

```diff
@@ -0,0 +1,129 @@
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        //SETS UP RECYCLERVIEW LAYOUT
+        mRecyclerView = (RecyclerView)findViewById(R.id.recycler);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this,
+                LinearLayoutManager.VERTICAL, false));
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent = new Intent(MainActivity.this, ShoppingCartActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        //** METHOD TO START TASK TO CHECK/ADD ITEMS TO DB***************
+        fillDataBase();
+
+    }
+
+    public void fillDataBase(){
+        final DatabaseHelper db = DatabaseHelper.getInstance(this);
+
+        if (mTask != null && mTask.getStatus() == AsyncTask.Status.RUNNING){
+            Toast.makeText(this, "Still adding data to the database. Please wait",
+                    Toast.LENGTH_LONG).show();
+        }else {
+            mTask = new AsyncTask<Void, Void, Void>() {
+                @Override
+                protected Void doInBackground(Void... voids) {
+                    if (db.getAllAsList().size() == 0) {
+                        db.populateGearTable();
+                    }
+                    return null;
+                }
+
+                @Override
+                protected void onPostExecute(Void aVoid) {
+                    super.onPostExecute(aVoid);
+                    mGearList = db.getAllAsList();
+                    mAdapter = new Adapter(mGearList,MainActivity.this);
+                    mRecyclerView.setAdapter(mAdapter);
+                }
+            };
+        mTask.execute();
+        }
+    }
+
+    @Override
+    protected void onNewIntent(Intent intent) {
+        setIntent(intent);
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent){
+        if (Intent.ACTION_SEARCH.equals(intent.getAction())){
+            String query = intent.getStringExtra(SearchManager.QUERY);
+
+            List<FWCGear> searchGear = DatabaseHelper.getInstance(this).searchGear(query);
+            mAdapter.replaceData(searchGear);
+        }
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+    }
+
+    //CREATES SEARCH BAR
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+
+        SearchManager searchManager = (SearchManager)getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView = (SearchView)menu.findItem(R.id.search_view).getActionView();
+        ComponentName componentName = new ComponentName(this, MainActivity.class);
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName));
+        return true;
```

Comment:
When the user closes the `SearchView` the list continues to show only the search results, and there's no way to make it go back to showing all items. To fix this, set an [`OnActionExpandListener`](https://developer.android.com/reference/android/view/MenuItem.OnActionExpandListener.html) and in the 'onMenuItemActionCollapse()` method of the listener update the RecyclerView to show all items again